core/src/wallet/algorand/AlgorandNetworks.cpp

```diff
@@ -59,6 +59,12 @@ namespace networks {
         getAlgorandNetworkParameters("algorand-testnet")
     });
 
+    const void checkAlgorandCurrency(const std::string &networkName) {
+        if (networkName != "algorand" && networkName != "algorand-testnet") {
+            throw make_exception(api::ErrorCode::UNSUPPORTED_CURRENCY, "Unsupported currency '{}'.", networkName);
+        }
+    }
+
```

Comment:
Such codes should be avoid in the future because :
- The function name `checkXXX` isn't explicit enough. Do the function returns a boolean ? Or throw an exception ? Or have side effects ? In our case is the second option but we need to read the code to be aware of.
- Both technical logic and error management are put at the same level of abstraction which is as I said in a previous comment a bad habit. 

So I suggest to create a `isAlgorandCurrency` which returns a boolean and the returned value can be pass as argument to an `assertCurrency` with a second argument indicating the error message. In that way we can reuse both helper functions and obviously create unit tests easily.

A second point is about how the currency is checked. There is a implicit correlation between free functions which is error-proned; each time a new currency is added we have to update the `checkAlgorandCurrency` and it's for sure something that will be forget by developers. 
I suggest to put the `AlgorandNetworkParameters` variables to an `unordered_map` and perform find operation in both cases. Like that we just have to add the new currency into that map and the behaviour of free functions will be modified as well.